app/models/course/assessment/question/programming.rb

```diff
@@ -174,10 +178,9 @@ def process_package
     elsif should_evaluate_package
       # For non-autograded questions, the attachment is not present
       evaluate_package if attachment
-    elsif is_codaveri_changed? || live_feedback_enabled_changed?
-      # Only when is_codaveri changed (no other setting), we recreate the codaveri
-      # problem to avoid attachment recreation and answers regrading
-      create_codaveri_problem if attachment
+    elsif (is_codaveri_changed? && is_codaveri?) || (live_feedback_enabled_changed? && live_feedback_enabled?)
```

Comment:
Need to check `is_synced_with_codaveri` here.

I believe the condition should be: `!is_synced_with_codaveri && (is_codaveri? || live_feedback_enabled?)`

Reply:
the condition of `is_codaveri_changed?` and `live_feedback_enabled_changed?` also needs to be checked as well, though I agree that `is_synced_with_codaveri` is essential so as we do not update Codaveri question in Codaveri side needlessly

The scenario is the following:
- we duplicate the question which has `is_codaveri` and `live_feedback_enabled` set to true
- the new object shall have both `is_codaveri` and `live_feedback_enabled` set to true, but that `codaveri_id` is nil and also `is_synced_with_codaveri` is false (the latter, I haven't included it, will include later within this PR. Also please note that the right thing is to switch `is_synced_with_codaveri` to false, since it's technically not synced with any codaveri question)
- we switch off either `is_codaveri` or `live_feedback_enabled`.

In your proposed gatekeep, it will create new codaveri question, even though we don't have any intention to do so.
The right one should be

```
!is_synced_with_codaveri && ((is_codaveri_changed? && is_codaveri?) || (live_feedback_enabled_changed? && live_feedback_enabled?))
```

I will add this line, and also modify the behavior during duplication as well